pkg/utils/gitrepo.go

```diff
@@ -934,19 +934,17 @@ func IsClusterAdmin(client client.Client, sub *appv1.Subscription, eventRecorder
 		}
 	}
 
-	// If subscription has cluster-admin:true and propagated from hub and cannot find the webhook, we know we are
-	// on the managed cluster so trust the annotations to decide that the subscription is from subscription-admin.
-	// But the subscription can also be propagated to the self-managed hub cluster.
+	// When the appsub has hosting-subscription annotation, the cluster-admin: true annotation is always respected, including 3 cases:
+	// 1: the appsub is on the managed cluster propagated by a hub appsub, there is no ocm-mutating-webhook found in the current cluster
+	// 2: the appsub is on the hub containing -local name suffix, the appsub is propagated by a hub appsub.
+	// 3: the appsub is on the hub not containing -local name suffix, the appsub is propagated by a hub sub of sub.
```

Comment:
for the regular hub appsub, since it doesn't have the hosting-subscription annotation, the cluster-admin anno is not respected. The cluster-admin annotation is generated to child appsub by hub appsub